modules/afamqp/afamqp.c

```diff
@@ -718,16 +718,16 @@ afamqp_dd_init(LogPipe *s)
   AMQPDestDriver *self = (AMQPDestDriver *) s;
   GlobalConfig *cfg = log_pipe_get_config(s);
 
-  if (!log_threaded_dest_driver_init_method(s))
-    return FALSE;
-
   if (self->auth_method == AMQP_SASL_METHOD_PLAIN && (!self->user || !self->password))
     {
       msg_error("Error initializing AMQP destination: username and password MUST be set!",
                 evt_tag_str("driver", self->super.super.super.id));
       return FALSE;
     }
 
+  if (!log_threaded_dest_driver_init_method(s))
+    return FALSE;
+
```

Comment:
hmmm... this part of the changeset is not so readable...
There is a hidden dependency - log_threaded_dest_driver_init_method depends on persist name (actually acquire queue method depends on that).
And currently only the relocation of this codechunk expresses this dependency.

It can goes in, but later we should express this dependency in a more formal manner.

Reply:
We discussed it in details with @MrAnno. This is the most bothering part for us too.

What do you think about the proposal to extract the queue initialization method from the init(), making the dependency more explicit? (That way we can also move most of the init methods back to their original place.)

Reply:
What if you extend the ctor with persist_name?
Or, create a `log_threaded_dst_driver_init_with_queue()` method?